profiles/api/oauth2.py

```diff
@@ -98,21 +98,34 @@ def _check() -> Response:
     @no_cache
     def _token() -> Response:
         if 'client_id' not in request.form:
+            LOGGER.error('Post request to /token without client_id specified')
             raise InvalidClient
 
+        LOGGER.info('Post request to /token from %s', request.form.get('client_id'))
+
         try:
             client = clients.find(request.form.get('client_id'))
         except KeyError as exception:
+            LOGGER.error('Invalid Client: %s not in list of known clients',
+                         request.form.get('client_id'))
             raise InvalidClient from exception
 
         redirect_uri = request.form.get('redirect_uri')
         if request.form.get('client_secret') != client.client_secret:
+            LOGGER.error('Invalid Client: client_secret not in request')
```

Comment:
maybe this error messages should use the precise class name e.g. `InvalidClient` to make it clear is something you can catch and/or search for in the code rather than just a log message

Reply:
ok, i'll convert for errors.